docs/examples/201/log-analytics-with-solutions-and-diagnostics/main.bicep

```diff
@@ -0,0 +1,124 @@
+param location string {
+    default: resourceGroup().location
+    allowed: [
+        'australiacentral' 
+        'australiaeast' 
+        'australiasoutheast' 
+        'brazilsouth'
+        'canadacentral' 
+        'centralindia' 
+        'centralus' 
+        'eastasia' 
+        'eastus' 
+        'eastus2' 
+        'francecentral' 
+        'japaneast' 
+        'koreacentral' 
+        'northcentralus' 
+        'northeurope' 
+        'southafricanorth' 
+        'southcentralus' 
+        'southeastasia'
+        'switzerlandnorth'
+        'switzerlandwest'
+        'uksouth' 
+        'ukwest' 
+        'westcentralus' 
+        'westeurope' 
+        'westus' 
+        'westus2' 
+    ]
+}
+
+param logAnalyticsWorkspaceName string = 'la-${uniqueString(resourceGroup().id)}'
+
+var vmInsights = {
+    name: 'VMInsights(${logAnalyticsWorkspaceName})'
+    galleryName: 'VMInsights'
+}
+var containerInsights = {
+    name: 'ContainerInsights(${logAnalyticsWorkspaceName})'
+    galleryName: 'ContainerInsights'
+}
+
+var environmentName = 'Production'
+var costCenterName = 'IT'
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2020-03-01-preview' = {
+    name: logAnalyticsWorkspaceName
+    location: location
+    tags: {
+        Environment: environmentName
+        CostCenter: costCenterName
+    }
+    properties: {
+        retentionInDays: 30
+        features: {
+            searchVersion: 1
+        }
+        sku: {
+            name: 'PerGB2018'
+        }
+    }
+}
+
+resource logAnalyticsWorkspaceDiagnostics 'Microsoft.OperationalInsights/workspaces/providers/diagnosticSettings@2017-05-01-preview' = {
+    name: '${logAnalyticsWorkspace.name}/Microsoft.Insights/diagnosticSettings'
+    properties: {
+        workspaceId: logAnalyticsWorkspace.id
+        logs: [
+            {
+                category: 'Audit'
+                enabled: true
+                retentionPolicy: {
+                    days: 7
+                    enabled: true
+                }
+            }
+        ]
+        metrics: [
+            {
+                category: 'AllMetrics'
+                enabled: true
+                retentionPolicy: {
+                    days: 7
+                    enabled: true
+                }
+            }
+        ]
+    }
+}
+
+resource solutionsVMInsights 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = {
+    name: '${vmInsights.name}'
+    location: location
+    dependsOn: [
+        logAnalyticsWorkspace
+    ]
+    properties: {
+        workspaceResourceId: logAnalyticsWorkspace.id 
+    }
+    plan: {
+        name: '${vmInsights}.Name'

```

Comment:
This looks incorrect. I think you want the `name` property from the `vmInsights` resource, in which case you would want `${vmInsights.name}` with `.name` inside of the `}`. That being said, this is another case where string interpolation is not required, so I'd recommend changing this to:

```
name: vmInsights.name
```

Reply:
updated to vmInsights.name